cirq-google/cirq_google/serialization/circuit_serializer.py

```diff
@@ -422,6 +446,19 @@ def deserialize(self, proto: v2.program_pb2.Program) -> cirq.Circuit:
                     deserialized_constants.append(circuit.freeze())
                 elif which_const == 'qubit':
                     deserialized_constants.append(v2.qubit_from_proto_id(constant.qubit.id))
+                elif which_const == 'operation_value':
+                    deserialized_constants.append(
+                        self._deserialize_gate_op(
+                            constant.operation_value,
+                            arg_function_language=arg_func_language,
+                            constants=proto.constants,
+                            deserialized_constants=deserialized_constants,
+                        )
+                    )
+                elif which_const == 'moment_value':
+                    # For now, just put in a place holder, since moments
+                    # will be deserialized later.
+                    deserialized_constants.append(None)
```

Comment:
Is this a bug that is being resolved? If so, can you add a test that would have have failed without this fix but passes now?

Reply:
The existing tests will catch it.  This was a preexisting bug, but it did not appear since the moments would only be at the end of the constants table.  In one of the tests, there are multiple moments and operations, which intersperses them in the list, so this test will fail if this is removed.